hudi-client/hudi-client-common/src/main/java/org/apache/hudi/execution/ParquetFileMetaToWriteStatusConvertor.java

```diff
@@ -0,0 +1,108 @@
+ */
+public class ParquetFileMetaToWriteStatusConvertor<T extends HoodieRecordPayload, I, K, O> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ParquetFileMetaToWriteStatusConvertor.class);
+  private final HoodieTable<T,I,K,O> hoodieTable;
+  private final HoodieWriteConfig writeConfig;
+  private final FileSystem fs;
+
+  public ParquetFileMetaToWriteStatusConvertor(HoodieTable<T, I, K, O> hoodieTable, HoodieWriteConfig writeConfig) {
+    this.hoodieTable = hoodieTable;
+    this.writeConfig = writeConfig;
+    this.fs = this.hoodieTable.getMetaClient().getFs();
+  }
+
+  /**
+   * This method generates writeStatus object from parquet file.
+   */
+  public WriteStatus convert(String parquetFile, String partitionPath,
+                             Map<String, Object> executionConfigs) throws IOException {
+    LOG.info("Creating write status for parquet file " + parquetFile);
+    WriteStatus writeStatus = (WriteStatus) ReflectionUtils.loadClass(this.writeConfig.getWriteStatusClassName(),
+        !this.hoodieTable.getIndex().isImplicitWithStorage(), this.writeConfig.getWriteStatusFailureFraction());
+    Path parquetFilePath = new Path(parquetFile);
+    writeStatus.setFileId(FSUtils.getFileId(parquetFilePath.getName()));
+    writeStatus.setPartitionPath(partitionPath);
+    generateHoodieWriteStat(writeStatus, parquetFilePath, executionConfigs);
+    return writeStatus;
+  }
+
+  /**
+   * This method generates HoodieWriteStat object and set it as part of WriteStatus object.
+   */
+  private void generateHoodieWriteStat(
+      WriteStatus writeStatus, Path parquetFilePath, Map<String, Object> executionConfigs) throws IOException {
+    HoodieWriteStat stat = new HoodieWriteStat();
+
+    // Set row count
+    ParquetMetadata parquetMetadata = ParquetFileReader.readFooter(this.fs.getConf(), parquetFilePath,
+        ParquetMetadataConverter.NO_FILTER);
+    List<BlockMetaData> blockMetaDataList = parquetMetadata.getBlocks();
+    long rowCount = blockMetaDataList.stream().mapToLong(BlockMetaData::getRowCount).sum();
+    stat.setNumWrites(rowCount);
+    stat.setNumInserts(rowCount);
+
+    // Set runtime stats
+    HoodieWriteStat.RuntimeStats runtimeStats = new HoodieWriteStat.RuntimeStats();
+    runtimeStats.setTotalCreateTime((long) executionConfigs.get("timeTaken"));
+    stat.setRuntimeStats(runtimeStats);
+
+    // File size
+    FileStatus parquetFileStatus = this.fs.getFileStatus(parquetFilePath);
+    long fileSize = parquetFileStatus.getLen();
+    stat.setFileSizeInBytes(fileSize);
+    stat.setTotalWriteBytes(fileSize);
+
+    stat.setFileId(writeStatus.getFileId());
+    stat.setPartitionPath(writeStatus.getPartitionPath());
+    stat.setPath(new Path(writeConfig.getBasePath()), parquetFilePath);
+    stat.setTotalWriteErrors(writeStatus.getTotalErrorRecords());
+    stat.setPrevCommit(String.valueOf(executionConfigs.get("prevCommit")));
+
+    writeStatus.setStat(stat);
+  }
```

Comment:
oh this is not populating any succ records. anyways, we can build something on top of this for insert_overwrite support